pkgs/by-name/em/emojipick/package.nix

```diff
@@ -0,0 +1,79 @@
+{
+  lib,
+  dmenu,
+  fetchFromGitHub,
+  installShellFiles,
+  libnotify,
+  python3,
+  rofi,
+  stdenvNoCC,
+  xclip,
+  # Boolean flags
+  emojipick-copy-to-clipboard ? true,
+  emojipick-print-emoji ? true,
+  emojipick-show-notifications ? true,
+  emojipick-use-rofi ? false,
+  # Configurable options
+  emojipick-font-family ? "Noto Color Emoji",
+  emojipick-font-size ? "18",
+
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "emojipick";
+  version = "20210127";
+
+  src = fetchFromGitHub {
+    owner = "thingsiplay";
+    repo = "emojipick";
+    rev = finalAttrs.version;
+    hash = "sha256-96u2wcuSnngiP2Ew2Zcs81pFpy7dZW84Tjt90z0bK84=";
+  };
+
+  nativeBuildInputs = [
+    installShellFiles
+  ];
+
+  buildInputs = [
+    libnotify
+    python3
+    xclip
+    (if emojipick-use-rofi then rofi else dmenu)
+  ];
+
+  dontConfigure = true;
+
+  dontBuild = true;
+
+  strictDeps = true;
+
+  # notify-send has to be patched in a bash file
+  postPatch = ''
+    substituteInPlace emojipick \
+      --replace "use_rofi=0" "use_rofi=${builtins.toString (lib.boolToInt emojipick-use-rofi)}" \
```

Comment:
"Converting true to 1 and false to nothing" remembers me that [Javascript madness](https://gist.github.com/JakeSidSmith/e38f661a7231a912d2d2d380352664da) things.
Plus the less than stellar Bash pitfalls, I think this is not a good idea.

Reply:
Haha fair. I do wonder how worth it is to have such a function in `lib`, would be great if you can show some other places where the function could be applied. If you think it's worth it, it should be a separate PR and include a simple test, see also `lib`'s [PR guidelines](https://github.com/NixOS/nixpkgs/tree/master/lib#pr-guidelines).

Reply:
I found a similar construction `oz = flag: if flag then 1 else 0;` on some packages that configure environment variables for Makefiles.

Further, I have found many makefiles out there that use 0-as-false-1-as-true  hack/abstraction, since Makefiles work with strings.

I will compile some survey.